contracts/Vault.vy

```diff
@@ -302,11 +302,15 @@ def _issueSharesForAmount(_to: address, _amount: uint256) -> uint256:
     return shares
 
 
-@external
-def deposit(_amount: uint256) -> uint256:
+@internal
+def _deposit(_sender: address, _amount: uint256) -> uint256:
     assert not self.emergencyShutdown  # Deposits are locked out
     assert self._totalAssets() + _amount <= self.depositLimit  # Max deposit reached
 
+    # Depositer should have funds
+    assert self.token.balanceOf(_sender) > 0
+    assert self.token.balanceOf(_sender) >= _amount
```

Comment:
This is inherently true if `self.token.transferFrom(_sender, self, _amount)` succeeds (so, not necessary)